oeps/oep-0011.rst

```diff
@@ -0,0 +1,299 @@
+   all possible libraries. It also supports intelligent bundling using
+   RequireJS Optimizer, which can package together all required libraries
+   into a single file for faster performance.
+
+7. **CSS should be generated using SASS**
+
+   Rationale: Sass is an extension of CSS that adds power and elegance to
+   the basic language. It makes the maintenance of large CSS files more
+   manageable though the ability to use variables, mixins, imports and more.
+   In particular, it makes theming possible with the ability to override
+   variables that define colors, fonts etc.
+
+   You can find out more about Sass with the official guide
+   `Sass: CSS with Superpowers`_.
+
+Open Questions
+==============
+
+.. _JavaScript frameworks:
+
+1. **Which more modern JavaScript framework should be supported?**
+
+   The JavaScript space is very fast moving as can be seen in
+   `The State of JavaScript 2016 Survey`_. There is a lot of momentum
+   around `React`_, and it appears to have many qualities that would
+   make it a good fit for edX. It is too early to make this call, but
+   plans are being made to evaluate its usage. See `React at edX`_ for
+   more details.
+
+2. **How should offline support be provided?**
+
+   For many mobile users, it would be very useful if they could continue
+   their Open edX courses while offline. This is true both for true mobile
+   apps as well as for responsive web applications. This is an area that
+   has not been explored much up to now, but which will only become more
+   important going forward.
+
+3. **Should Universal JavaScript be considered?**
+
+   There are many companies that are seeing performance benefits by
+   pre-rendering their dynamic client-side applications on the server first.
+   The most common approach is to run the same JavaScript code both in the
+   browser and using Node on the server. This requires the use of a JavaScript
+   framework that does not perform direct manipulation of the DOM, but instead
+   works with an abstraction of the desired HTML. For example, Netflix is
+   seeing great performance gains by using `React`_ in this way, as can be
+   seen in the blog post `Making Netflix Faster`_.
+
+4. **Where do web components fit with Open edX applications?**
+
+   Web Components is a W3C recommendation that allows developers to create
+   custom HTML elements for use within their web applications. The
+   `Polymer Project` provides a library that allows web components to be
+   built in most modern browsers. It seems that there is a useful overlap
+   between web components and Open edX's `XBlocks`_
+
+5. **Which build tools and asset pipeline should be used?**
+
+   There are already a variety of technologies used within Open edX,
+   none of which is a clear winner. Django Pipeline is used by
+   edx-platform, Django Compressor is used by some of the Ecommerce
+   applications, and Webpack is prevalent in pure front-end implementations.
```

Comment:
We'll need to reconcile the endorsement of RequireJS Optimizer above with the endorsement of Webpack here.